server/midddleware/metrics.go

```diff
@@ -33,22 +34,34 @@ import (
 
 func getOkMethodTags(ctx context.Context, methodName string, methodType string) map[string]string {
 	tags := metrics.GetPreinitializedTagsFromFullMethod(methodName, methodType)
-	namespace, err := GetNamespace(ctx)
+	namespace, err := request.GetNamespace(ctx)
 	if err != nil {
-		ulog.E(err)
+		switch err.(type) {
+		case *request.NamespaceNotFound:
+			tags["namespace"] = "unknown"
+		default:
+			ulog.E(err)
+		}
+	} else {
+		tags["namespace"] = namespace
 	}
-	tags["namespace"] = namespace
 	return tags
 }
 
 func getErrorMethodTags(ctx context.Context, fullMethod string, methodType string, errSource string, errCode string) map[string]string {
 	metaData := metrics.GetGrpcEndPointMetadataFromFullMethod(fullMethod, methodType)
 	tags := metaData.GetSpecificErrorTags(errSource, errCode)
-	namespace, err := GetNamespace(ctx)
+	namespace, err := request.GetNamespace(ctx)
 	if err != nil {
-		ulog.E(err)
+		switch err.(type) {
+		case *request.NamespaceNotFound:
+			tags["namespace"] = "unknown"
```

Comment:
@JigarJoshi I guess you are already adding it in auth interceptor, do we need to do it here?

Reply:
this can be avoided here. auth namespace will place `unknown` if the auth fails and namespace is supplied in `""`

Reply:
in all other cases, there will always be a namespace.